src/components/DateSelector/index.js

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { makeStyles } from '@material-ui/core/styles';
+import DateFnsUtils from '@date-io/date-fns';
+import {
+  MuiPickersUtilsProvider,
+  KeyboardDatePicker,
+} from '@material-ui/pickers';
+import { styles } from './style';
+
+const useStyles = makeStyles(styles);
+
+export const DateSelector = ({
+  name,
+  label,
+  defaultValue,
+  onFocus,
+  helperText,
+  error,
+  disabled,
+}) => {
+  const [selectedDate, setSelectedDate] = useState(null);
+  const classes = useStyles();
+
+  useEffect(() => {
+    if (defaultValue) {
+      if (error) {
+        setSelectedDate(null);
+      } else {
+        setSelectedDate(defaultValue);
+      }
+    }
+  }, []);
```

Comment:
you currently have this line:
```js
const [selectedDate, setSelectedDate] = useState(null);
```

so it means by default your `selectedDate` already `null`

then wouldn't be better to have this condition like so:
```js
if (defaultValue && !error) setSelectedDate(defaultValue):
```
?